frontend/src/scenes/dashboard/newDashboardForm.ts

```diff
@@ -0,0 +1,78 @@
+import { kea } from 'kea'
+import type { newDashboardFormType } from './newDashboardFormType'
+import { DashboardRestrictionLevel } from 'lib/constants'
+import { DashboardType } from '~/types'
+import api from 'lib/api'
+import { teamLogic } from 'scenes/teamLogic'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+import { dashboardsModel } from '~/models/dashboardsModel'
+
+export interface NewDashboardForm {
+    name: string
+    show: boolean
+    useTemplate: string
+    restrictionLevel: DashboardRestrictionLevel
+}
+
+const defaultFormValues: NewDashboardForm = {
+    name: '',
+    show: true,
+    useTemplate: '',
+    restrictionLevel: DashboardRestrictionLevel.EveryoneInProjectCanEdit,
+}
+
+export const newDashboardForm = kea<newDashboardFormType<NewDashboardForm>>({
```

Comment:
Is this `fooForm` logic pattern something we should be doing for all forms?

Reply:
I'm not yet sure. I want to refactor a few more forms to figure this out.

Reply:
Historically when I've separated a form into its own logic (I had a function `createForm(props) => kea(convertProps(props))` in my last projects), it would be possible to end up in two-way dependencies where `userForm` and `userLogic` want to read or write each others data in weird ways.

So I guess, "it depends" is a safe answer... 🤔 

I'm not sure if the "*Form" naming is useful at the end of the day, and perhaps it should just still be `newDashboardLogic`. 

What do you think?

Reply:
Makes sense. I'll rename.